src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,38 @@
 package mate.academy;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final ConcurrentHashMap<Integer, EventListener> listeners = new ConcurrentHashMap<>();
 
+    public void registerListener(EventListener listener) {
+        listeners.put(listener.hashCode(), listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener.hashCode());
     }
 
     public void notifyEvent(Event event) {
-
+        List<Future> futures = new ArrayList<>();
```

Comment:
Consider using a parameterized type for `Future` to ensure type safety. For example, `List<Future<Void>> futures = new ArrayList<>();`.